functions/src/index.ts

```diff
@@ -2,32 +2,59 @@ import * as functions from "firebase-functions";
 import { store } from "./database";
 
 export const save = functions.https.onRequest((request, response) => {
-  const event = {
-    // Client-provided values
-    uid: request.body.uid,
-    client: request.body.client,
-    key: request.body.key,
-    value: request.body.value,
+  const body = request.body;
+  console.log(`Body = ${JSON.stringify(body)}`);
 
-    // Optional (TODO: make it required like the other fields)
-    sent: request.body.sent,
+  const event = Object.assign({}, body, {
+    receivedAt: new Date().toISOString()
+  });
 
-    // Additional values
-    received: new Date().toISOString()
-  };
+  const requiredFields: string[] = [
+    // A key used to identify the value in this event (it need not be unique).
+    // For example, this could be the type of the event.
+    'key',
+
+    // The data that the client wants to store in the database.
+    'value',
+
+    // An identifier for the user associated with this event.
+    'uid',
+
+    // An identifier for the client issuing this request (e.g., the app
+    // version). Think of this as the client user-agent.
+    'clientVersion',
+
+    // Timestamp for when the client generated the event. NOTE: This should be
+    // when the event actually occurred on the client and NOT when it was sent
+    // to the server.
+    'timestamp',
```

Comment:
I think it might be nice for the two timestamps to have more "symmetrical" names, e.g., `clientTimestamp` and `serverTimestamp`. (I'm also okay with the original field names, `sent`/`received`.)

(In general, I want to nudge us to use the server timestamp; the client timestamp should be used primarily for debugging, because it can be way off due to clock skew.)